cpp/CPP14Parser.g4

```diff
@@ -359,21 +364,27 @@ typeSpecifierSeq: typeSpecifier+ attributeSpecifierSeq?;
 trailingTypeSpecifierSeq:
 	trailingTypeSpecifier+ attributeSpecifierSeq?;
 
+simpleTypeLengthModifier:
+	Short
+	| Long;
+	
+simpleTypeSignednessModifier:
+	Unsigned
+	| Signed;
+
 simpleTypeSpecifier:
 	nestedNameSpecifier? theTypeName
 	| nestedNameSpecifier Template simpleTemplateId
-	| Char
-	| Char16
-	| Char32
-	| Wchar
+	| simpleTypeSignednessModifier
+	| simpleTypeSignednessModifier? simpleTypeLengthModifier+
```

Comment:
This will catch `short long short long short long`. Is that possible?

Reply:
It certainly will.  But should that be limited by the grammar itself or should that be sorted out by higher level code? I am new to this and so I not quite sure where to draw the line.

Reply:
I really am not sure for this, I prefer being able to go with a more conservative approach.

Reply:
It should be limited by grammar if possible.

Reply:
It is better than it was (ie short int, long int and such declarations as unsigned long were not captured).  We can follow up with something that restricts it to specific patterns of unsigned, signed, short and long later.